WTWSMS/decisions/WtWSMS_diploaction_decisions.txt

```diff
@@ -96,4 +96,79 @@ targetted_decisions = {
 			factor = 1
 		}
 	}
+	
+	# TODO: ask_to_become_foederati_emperor  (Empire->barbarians)
+	
+	become_foederati_barbarian = { # barbarians->Empire
+		only_independent = yes
+		ai_check_interval = 12
+
+		potential = {
+			AND = {
+				primary_title = { higher_tier_than = DUKE }
+				has_romance_culture_trigger = yes
+			}
+		}
+		
+		from_potential = {
+			independent = yes
+			mercenary = no
+			holy_order = no
+			is_landed = yes
+		}
+		
+		allow = {
+			FROM = {
+				show_scope_change = no
+				war = no
+				OR = {
+					custom_tooltip = { #Borders the Roman Empire
+						text = become_foederati_barbarian_allow_borders_roman_realm_tt
+						any_neighbor_independent_ruler = {
+							AND = {
+								primary_title = { higher_tier_than = DUKE }
+								has_romance_culture_trigger = yes
+							}
+						}
+					}
+				}
+				NOT = {
+					is_tributary = yes
+				}
+				NOT = { tier = emperor }
+				custom_tooltip = {
+					text = become_foederati_barbarian_cooldown_tt
+					NOT = { has_character_modifier = become_foederati_timer }
+				}
+				prisoner = no
+				NOT = { trait = incapable }
+				is_inaccessible_trigger = no
+			}
+		}
+		
+		effect = {
+			make_tributary = {
+				who = FROM
+				tributary_type = foederati
+			}
```

Comment:
Is your intent to change this to event later on, or do you personally feel done with expanding this as it is?

Reply:
The negotiations have to be written, but the current decision should work well, I did some testing.